src/org/exist/xquery/functions/fn/LoadXQueryModule.java

```diff
@@ -0,0 +1,214 @@
+                            "entry for each public function declared in the library module, except that when two functions have " +
+                            "the same name (but different arity), they share the same entry. The key of the entry is the name of the " +
+                            "function(s), as an xs:QName value; the associated value is a map A. This map (A) contains one entry for each " +
+                            "function with the given name; its key is the arity of the function, as an xs:integer value, and its associated " +
+                            "value is the function itself, as a function item. The function can be invoked using the rules for dynamic " +
+                            "function invocation.")
+    );
+
+    public final static FunctionSignature LOAD_XQUERY_MODULE_2 = new FunctionSignature(
+            new QName("load-xquery-module", Function.BUILTIN_FUNCTION_NS, FnModule.PREFIX),
+            "Provides access to the public functions and global variables of a dynamically-loaded XQuery library module.",
+            new SequenceType[] {
+                    new FunctionParameterSequenceType("module-uri", Type.STRING,
+                            Cardinality.EXACTLY_ONE, "The target namespace of the module"),
+                    new FunctionParameterSequenceType("options", Type.MAP,
+                            Cardinality.EXACTLY_ONE, "Options for loading the module")
+            },
+            new FunctionReturnSequenceType(
+                    Type.MAP,
+                    Cardinality.EXACTLY_ONE,
+                    "a map with two entries: 1) 'variables': a map with one entry for each public global variable declared in " +
+                            "the library module. The key of the entry is the name of the variable, as an xs:QName value; the " +
+                            "associated value is the value of the variable; 2) 'functions': a map which contains one " +
+                            "entry for each public function declared in the library module, except that when two functions have " +
+                            "the same name (but different arity), they share the same entry. The key of the entry is the name of the " +
+                            "function(s), as an xs:QName value; the associated value is a map A. This map (A) contains one entry for each " +
+                            "function with the given name; its key is the arity of the function, as an xs:integer value, and its associated " +
+                            "value is the function itself, as a function item. The function can be invoked using the rules for dynamic " +
+                            "function invocation.")
+    );
+
+    public final static StringValue OPTIONS_LOCATION_HINTS = new StringValue("location-hints");
+    public final static StringValue OPTIONS_XQUERY_VERSION = new StringValue("xquery-version");
+
```

Comment:
(2) is difficult because eXist does neither support external variables in library modules nor default values. I tried to add both, so there will be a new commit once the test suite passes.